src/main/java/com/gruzini/tennistico/models/dto/SetDto.java

```diff
@@ -18,21 +20,22 @@
 
    @AssertTrue
    public boolean isSetValid() {
-
-      //normal scenario
-      if (hostScore == 6 && guestScore <= 4) {
-         return true;
-      }
-      if (guestScore == 6 && hostScore <= 4) {
-         return true;
+      if(!isNull(hostScore) && !isNull(guestScore)){
+         //normal scenario
+         if (hostScore == 6 && guestScore <= 4) {
+            return true;
+         }
+         if (guestScore == 6 && hostScore <= 4) {
+            return true;
+         }
+         //tiebreak scenario
+         if (hostScore == 7 && guestScore == 6 || hostScore == 6 && guestScore == 7) {
+            return true;
+         }
+         //advantage scenario
+         return hostScore == 7 && guestScore == 5 || hostScore == 5 && guestScore == 7;
+         //other scenarios are invalid
       }
-      //tiebreak scenario
-      if (hostScore == 7 && guestScore == 6 || hostScore == 6 && guestScore == 7) {
-         return true;
-      }
-      //advantage scenario
-      return hostScore == 7 && guestScore == 5 || hostScore == 5 && guestScore == 7;
-      //other scenarios are invalid
-
+      return true;
```

Comment:
Teraz jak są nullami, to returnuje 'true'?

Reply:
Tak, i tego w sumie chcemy. Możemy podać tylko wynik jednego setu, pozostałe są nullami, więc słabo by było jakby nie przeszło walidacji. Co nie zmienia faktu, że nadal musisz podać pola, które są required, czyli cały set musisz wypełnić.